lstchain/scripts/onsite/onsite_create_calibration_file.py

```diff
@@ -235,16 +235,20 @@ def main():
         log_file = f"{output_dir}/log/{output_name}.log"
         print(f"\n--> Log file {log_file}")
 
+        yes = False
         if os.path.exists(output_file):
-            if query_yes_no(">>> Output file exists already. Do you want to remove it?"):
+            if os.getenv('SLURM_JOB_ID') is None:
```

Comment:
I think this variable can be only interpreted inside a script launched with sbatch. Here it will be always None.

Reply:
indeed. This is the current johb id, not the one of a job launches using `sbatch` and friends

Reply:
yes, and it will be not None when the script is running in batch (this is the idea)

Reply:
But the check is to make sure that in SLURM, the script is not asking for user input.

However, there is a better way: adding the conventional `--yes` option that will prevent dialogs and answers them so that the script continues automatically.

Reply:
?

Reply:
It is convention that many programs follow, to add a `-y, --yes` command line option that will prevent any dialogs.

e.g. like this:

```
parser.add_argument('-y', '--yes', help='Do not ask interactively for permissions, assume yes')
```

and then for example later:

```
overwrite = args.yes or query_yes_no("Overwrite?")
```

Reply:
The sbatch script can then add the `--yes` option to prevent asking for any user input

Reply:
I see, never met this option. But the present implementation has the advantage to automatically switch off the dialog in case of batch (without recalling to use the option),  what is the problem with it?

Reply:
Both ways are fine. Please note that there are other (onsite) scripts with the same problem though.

Reply:
right

Reply:
I added also the yes option, it can indeed be useful in case one want to directly overwrite in the interactive case